### PR TITLE
[CP Staging][CRITICAL] Implement <WorkspaceWorkflowsApprovalsEditPage /> for Editing Workflow

### PR DESCRIPTION
**Pull request by blazejkustra**
### Details

[Link to the doc](https://docs.google.com/document/d/12X-hsNo9BBRcG5NSS1JzyjWD1w-iM0uciQJSD1i5mZc/edit). 

#### UI changes

image

image

### Fixed Issues
$ https://github.com/Expensify/App/issues/45953
PROPOSAL: N/A

### Tests
- [x] Open Workflows page (it has to be turned on in `More features` settings page)
- [x] Locally edit `canUseWorkflowsAdvancedApproval` function to return true (or use `canUseWorkflowsAdvancedApproval` beta)
- [x] Click on the 'Add approval workflow' button, and add an approval workflow (or more)
- [x] After creating one or more workflows click on the approval workflow section to edit an existing workflow
  - You should be able to change members, update approvers and remove workflows
  - Make sure to test circular references (one workflow: A - > B, second workflow B -> A), approvers assigned to multiple workflows

### Offline tests
While offline make sure Offline Pattern B is used, approval workflow sections should be greyed out after removing/updating a workflow.

### QA Steps
Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android.webm](https://github.com/user-attachments/assets/a536a32a-5fa4-4d22-8ed1-c8f1d4d506b1)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[android-web.webm](https://github.com/user-attachments/assets/1fe25196-60c3-4eb5-9efa-f9852c8b037d)

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/73b2121d-0892-4b72-b958-1516c5d07554

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/22f03faa-c1bc-4903-be39-250e4d69d012

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/f43e9326-232b-482f-99a9-b2a80567ef65

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/74d39596-da2b-4c6b-94c8-2623996b9c69

</details>

### CODE REVIEW

File: src/libs/WorkflowUtils.ts
Comment (blazejkustra) [inline src/libs/WorkflowUtils.ts:167]: The idea behind this function changed a bit, now it's no longer needed to spread current `employeeList` (with role, previous submitsTo, forwardsTo etc).
- For members set `submitsTo` to the first approver. For `remove` type, `submitsTo` is set to an empty string, which is handled on the backend, `submitsTo` is set to the default approver (policy.approver or if empty policy.owner)
- For approvers set `forwardsTo` to the next approver in the list (or clear it with an empty string for `remove` type)

cc @DylanDylann

File: src/libs/actions/Workflow.ts
Comment (blazejkustra) [inline src/libs/actions/Workflow.ts:188]: We have to use owner as a fallback (approver is optional) cc @DylanDylann

File: src/libs/actions/Workflow.ts
Comment (blazejkustra) [inline src/libs/actions/Workflow.ts:329]: This is a trick that is later used whenever `validateApprovalWorkflow` is called in components, `approvalWorkflow` is narrowed down to `ApprovalWorkflowOnyxValidated` type

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (blazejkustra) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:69]: We have to support linking to an edit page, which is a bit tricky as we have to calculate all `workflows` on the go first. @DylanDylann if you have a better idea how to load it based on the `route.params.firstApproverEmail`, I'm open for suggestions

**Comment by tgolen** [comment]
Yeah, the default approver had to be something I changed on the backend. I opened up https://github.com/Expensify/App/issues/47344 to handle that and I have a PR in review for it now. It should be deployed by tomorrow!

**Review by tgolen — DISMISSED** [review]
Code is looking good! It tests pretty well too with the recent backend changes. I think the final case is just making sure the default approver can be updated.

File: src/components/ApprovalWorkflowSection.tsx
Comment (DylanDylann) [inline src/components/ApprovalWorkflowSection.tsx:22]: @blazejkustra Why do you make this change? I think `onPress` function could be handled in ApprovalWorkflowSection.

**Comment by DylanDylann** [comment]
@blazejkustra BUG: When all members have `submitTo`, the edit page becomes empty

Uploading Screen Recording 2024-08-14 at 11.12.56.mov…

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:37]: @blazejkustra Why do we need to create this state? I think using `approvalWorkflow` (from Onyx) is enough

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:84]: We can see this case if trying to edit the approver email in the URL. I think we should display notFoundPage in this case

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:87]: @blazejkustra  With this condition, there are some cases that `approvalWorkflow` is not updated 


Whoops, Can't upload a video on GH today. I will send to you a video via Slack

Uploading Screen Recording 2024-08-14 at 11.45.12.mov…

File: src/components/ApprovalWorkflowSection.tsx
Comment (blazejkustra) [reply src/components/ApprovalWorkflowSection.tsx:22]: I prefer this section to be controlled by the parent, so no business logic is built into it

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:87]: Removed 👌

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:37]: It is used for both updating and removing a workflow:
- Remove: When user is editing a workflow and decides to remove it we can't rely on `approvalWorkflow` which was changed by the user, we need to remove the original workflow which is saved on the backend (in this case `initialApprovalWorkflow` - maybe the name is misleading?)

```ts
// Remove the approval workflow using the initial data as it could be already edited
Workflow.removeApprovalWorkflow(route.params.policyID, initialApprovalWorkflow);
```

- Update: We need to calculate what members were removed from the workflow, which is not trivial with just `approvalWorkflow`, see this logic:
```ts
const membersToRemove = initialApprovalWorkflow.members.filter((initialMember) => !approvalWorkflow.members.some((member) => member.email === initialMember.email));
Workflow.updateApprovalWorkflow(route.params.policyID, approvalWorkflow, membersToRemove);
```

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx]: Fixed 👌

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (DylanDylann) [reply src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:36]: @blazejkustra Should we move `currentApprovalWorkflow` outside the useEffect and use it instead of initialApprovalWorkflow?

**Comment by DylanDylann** [comment]
The strange bug: Not found page when removing the first approver and add another approver again 

Uploading Screen Recording 2024-08-14 at 13.55.40.mov…

File: src/libs/actions/Workflow.ts
Comment (DylanDylann) [inline src/libs/actions/Workflow.ts:173]: @blazejkustra What problem here?

File: src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx:162]: I think we should add a new props called `shouldDisplayDeleteButton` and pass this prop from parent component

```
const shouldDisplayDeleteButton = approvalWorkflow.action === EDIT && !approvalWorkflow.isDefault
```

File: src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx:36]: Let's remove policyID props and use policy.ID

Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx:36]: policy.policyID can be undefined which is making this so much harder, will leave it as is for now.

File: src/libs/actions/Workflow.ts
Comment (blazejkustra) [reply src/libs/actions/Workflow.ts:173]: This 😢 https://github.com/Expensify/App/issues/47344#issuecomment-2287895193

File: src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx
Comment (blazejkustra) [reply src/pages/workspace/workflows/approvals/ApprovalWorkflowEditor.tsx:162]: Either way I'll provide both `removeApprovalWorkflow` and `approvalWorkflow` props from the parent, so let's call it a NAB and leave as is

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

**Review by rlinoz — APPROVED** [review]

**Comment by rlinoz** [comment]
Asked for CP here: https://expensify.slack.com/archives/C07J32337/p1723635196903979

**Comment by lanitochka17** [comment]
### Approve button not displayed for updated approver
## Reproducible in production: 
Yes

## Expected Result: 
The approve button is displayed for the updated approver

## Actual Result:
There is no approve button for the updated approver, and the expense preview is empty

## Actions Performed:
precondition: as admin create an workspace, invite a submitter and approver, and set an approval workflow
1. Log in as submitter and submit some expenses
2. Log in as admin
3. Navigate to the workspace settings and invite another approver
4. Change the approver to the newly invited one in the approval workflow
5. Log in as the updated approver

## Workaround:
Unknown

## Notes/Images/Video:


https://github.com/user-attachments/assets/0e31f66f-bfeb-4ace-a797-4683e55541e9

**Comment by lanitochka17** [comment]
> ### Approve button not displayed for updated approver
> ## Reproducible in production:
> Yes
> 
> ## Expected Result:
> The approve button is displayed for the updated approver
> 
> ## Actual Result:
> There is no approve button for the updated approver, and the expense preview is empty
> 
> ## Actions Performed:
> precondition: as admin create an workspace, invite a submitter and approver, and set an approval workflow
> 
> 1. Log in as submitter and submit some expenses
> 2. Log in as admin
> 3. Navigate to the workspace settings and invite another approver
> 4. Change the approver to the newly invited one in the approval workflow
> 5. Log in as the updated approver
> 
> ## Workaround:
> Unknown
> 
> ## Notes/Images/Video:
>  Bug6572030_1723643861881.Recording__681.mp4

**Comment by rlinoz** [comment]
Is the above not happening anymore? 🤔

**Comment by Beamanator** [comment]
FYI I believe this was deployed to prod yesterday, from this checklist - #47356

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (DylanDylann) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:63]: Come from https://github.com/Expensify/App/issues/51282, we need to make sure the modal is dismissed before the data is removed

File: src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx
Comment (jjcoffee) [inline src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx:73]: Coming from https://github.com/Expensify/App/issues/68497. After switching to advanced approval mode in OD, the `policy?.approver` is an empty string, which causes an incorrect default approver check here:

https://github.com/Expensify/App/blob/161b475121c7bcdff56e51dcee69e4d17f05de89/src/libs/WorkflowUtils.ts#L131

We should use `getDefaultApprover` instead, like here:

https://github.com/Expensify/App/blob/62d46030d586497f4c662c37dc80df8fa372237d/src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx#L110-L115

**Pull request merged**
